packages/xo-web/src/xo-app/proxies/connect-proxy-modal.js

```diff
@@ -0,0 +1,55 @@
+import _ from 'intl'
+import Component from 'base-component'
+import React from 'react'
+import SingleLineRow from 'single-line-row'
+import { Col, Container } from 'grid'
+import { Input as DebounceInput } from 'debounce-input-decorator'
+
+export default class ConnectProxyModal extends Component {
+  state = {
+    authenticationToken: undefined,
+    name: undefined,
+    address: undefined,
+    vmUuid: undefined,
+  }
+  get value() {
+    return this.state
+  }
+
+  render() {
+    const { address, authenticationToken, name, vmUuid } = this.state
+    return (
+      <Container>
+        <SingleLineRow className='mt-1'>
+          <Col size={6}>{_('proxyAuthToken')}</Col>
+          <Col size={6}>
+            <DebounceInput
+              className='form-control'
+              onChange={this.linkState('authenticationToken')}
+              value={authenticationToken}
+            />
+          </Col>
+        </SingleLineRow>
+        <SingleLineRow className='mt-1'>
+          <Col size={6}>{_('name')}</Col>
+          <Col size={6}>
+            <DebounceInput className='form-control' onChange={this.linkState('name')} value={name} />
+          </Col>
+        </SingleLineRow>
+        <SingleLineRow className='mt-1'>
+          <Col size={6}>{_('address')}</Col>
+          <Col size={6}>
+            <DebounceInput className='form-control' onChange={this.linkState('address')} value={address} />
+          </Col>
+        </SingleLineRow>
+        {_('or')}
```

Comment:
There is a use case for registering a proxy with only the VM UUID but this is not the main one we are trying to answer here.

Also, if possible, it's always preferable to have the VM UUID (even when specifying the address), because it allows XO to correctly handle the license associated with this proxy and it's also better supported support in the UI and for the upgrade system.

The UI should reflect this.

Reply:
So adding an information message in the modal explaining VM UUID is not mandatory when specifying the address but recommended can be a good solution i guess?

Reply:
Why not, my main beef is that in the current version, these fields appear exclusive.

Maybe, as a first step, you can require the address and recommend the VM UUID.

Also, FYI, no check is done on the `vmUuid` field, unlike the `vm` field for which we check that the value really match to an existing VM (but obviously it does not work in this specific case, where the proxy appliance is in a pool not connected to XO).

Reply:
I like what you're proposal :+1: